<a href="https://colab.research.google.com/github/innouguru/flyrank_ml_track_1/blob/main/work/notebooks/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

# 1. My Lane

## Chosen Lane
**Refresh / Content Opportunity Scoring**

## Why I Chose This Lane

I chose the **Refresh / Content Opportunity Scoring** lane because it addresses a practical business problem that many organizations face: deciding which content should be reviewed first when time and resources are limited.

Large websites often contain thousands of pages, making it unrealistic for editors to manually evaluate every page. Instead of attempting to predict whether a page will become successful, this project focuses on helping content teams prioritize which pages deserve human attention based on observable search and engagement signals.

This lane also provides an opportunity to develop a complete machine learning workflow throughout the internship. It combines exploratory data analysis, feature engineering, baseline scoring, supervised learning, model evaluation, and explainable recommendations. The final objective is not to replace human judgment, but to support better and more consistent decision-making.

## 2. The question: decision, action, cost of a wrong call

---
## The Question

**Which content pages should be prioritized for review based on their search and engagement signals?**

## Decision

Given limited editorial capacity, which pages should the content team review first?

---

## Action

Editors can review the highest-ranked pages first and decide whether improvements are necessary.

---

## Cost of a Wrong Call

A false positive means editors spend valuable time reviewing pages that were already performing well or did not require immediate attention.

A false negative means genuinely important pages may continue losing visibility or engagement because they were not identified early enough.

Both types of mistakes reduce the effectiveness of limited editorial resources.



In [8]:
# Cloning my repository
!git clone https://github.com/innouguru/flyrank_ml_track_1.git

fatal: destination path 'flyrank_ml_track_1' already exists and is not an empty directory.


In [9]:
import pandas as pd

df = pd.read_csv(
    "flyrank_ml_track_1/data/raw/content_refresh_anonymized.csv"
)

df.head()

,content_id,client_id,search_volume,competition,competition_level,cpc,content_type,main_intent,word_count,char_count,...,char_count_tier,ctr,avg_position,engagement_rate,scroll_rate,ai_traffic_pct,impression_tier,position_tier,trend_direction,trend_pct
0,content_304f48230142,client_f369cb89fc,10.0,0.67,HIGH,2.05,keyword article,transactional,3221.0,20457.0,...,15000-25000,0.76,10.6,5.88,4.55,0.0,good,striking,down,-41.4
1,content_a1fb4e703a9e,client_4e07408562,90.0,0.01,LOW,0.05,keyword article,informational,2481.0,15562.0,...,15000-25000,0.05,20.3,0.00,10.00,0.0,good,page_3_5,down,-57.7
2,content_9aa793d4d895,client_7f2253d7e2,0.0,0.00,LOW,0.00,keyword article,informational,3515.0,23643.0,...,15000-25000,0.09,36.5,0.00,28.57,0.0,good,page_3_5,down,-60.9
3,content_331d6c4de07b,client_19581e27de,10.0,0.00,LOW,0.00,keyword article,commercial,NaN,NaN,...,NaN,0.49,6.2,1.28,3.45,0.0,good,page_1,stable,-13.8
4,content_d99b7a2d90ca,client_3fdba35f04,0.0,0.00,LOW,0.00,keyword article,informational,2803.0,17469.0,...,15000-25000,0.13,44.0,0.00,24.29,0.0,good,page_3_5,down,-34.7


In [10]:
total_pages = len(df)
declining_pages = (df["trend_direction"] == "down").sum()
declining_pct = declining_pages / total_pages * 100

avg_impressions = df["impressions_90d"].mean()
avg_age = df["content_age_days"].mean()

print(f"Total pages: {total_pages:,}")
print(f"Declining pages: {declining_pages:,} ({declining_pct:.1f}%)")
print(f"Average impressions (90d): {avg_impressions:,.0f}")
print(f"Average content age: {avg_age:.0f} days")

Total pages: 30,000
Declining pages: 16,262 (54.2%)
Average impressions (90d): 5,200
Average content age: 256 days


## 3. Quick look at the data (2-3 real numbers)

Before selecting a research direction, I explored the starter dataset to understand whether this problem is suitable for machine learning.

The dataset contains **30,000 content pages**, providing a sufficiently large sample for exploratory analysis.

Approximately **54.2% of the pages (16,262 pages)** are labelled as having a downward trend. This indicates that identifying which declining pages deserve attention is a practical and relevant business problem.

The average page received approximately **5,200 impressions during the previous 90 days**, although traffic varies considerably across pages. Since some pages contribute far more visibility than others, reviewing every page equally would not be an efficient use of editorial resources.

The average content age is approximately **256 days**, suggesting that many pages have existed long enough for refresh decisions to be meaningful.

These observations support the choice of the Refresh / Content Opportunity Scoring lane because they show that the dataset contains enough variation in traffic, trends, and content maturity to prioritize review decisions.

## 4. Careful words: what I can and can't claim

This notebook presents an initial framing of a machine learning problem rather than a completed predictive system.

The exploratory statistics presented here describe patterns observed in the available dataset and should not be interpreted as evidence of causation. A page that is identified as a high-priority review candidate is not guaranteed to improve after being updated.

The goal of this project is to support editorial decision-making by ranking pages that may deserve review based on observable search and engagement signals. Final decisions should continue to involve human judgment, and any improvements should ideally be evaluated using future observations or controlled experiments.

## Self-check

Before you submit, confirm each line honestly:

- [x] Every section above is filled — markdown thinking AND the code that backs it
- [x] The notebook runs top to bottom with no errors (Runtime → Run all)
- [x] No client names, URLs, or private queries anywhere
- [x] My claims use careful words: observed, measured, directional, decision-support
- [x] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.